### PR TITLE
ticket-1413/SNB-Work-Unstable

### LINKED ISSUE #1413: Processing chain data work unstable
**Versions**
Skaled: 3.16.0-beta.3(4)
IMA: 1.3.5-beta.0
IMA contract mainnet: ima.1.3.4-rc.0( for both version 1.3.5-beta.0)
Skale Manager : 1.9.2-stable.0

**Environment**
20 active nodes on the network
4 schains up and running 

**Preconditions**
Update process to release versions 

**Steps to Reproduce/Description**
Observe logs after update 

**Expected behavior**
During init of IMA agent SNB should collect all information about other schain each hour: 
2023-01-23 02:39:41.749: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 03:39:43.613: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 04:39:45.392: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 05:39:47.213: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server

**Actual result** 
In some cases SNB init failed, see more logs below. 
2023-01-23 05:39:47.213: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 06:39:49.155: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 08:39:52.559: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 09:39:54.263: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
Init failed at 07:39 
NOTE: Not related to connect ECONNREFUSED

[SNBFail.txt](https://github.com/skalenetwork/IMA/files/10482018/SNBFail.txt)
[SNBSuccess.txt](https://github.com/skalenetwork/IMA/files/10482019/SNBSuccess.txt)

```[tasklist]
### Tasks
- [ ] https://github.com/skalenetwork/IMA/issues/1595
```

### PR DESCRIPTION
**Pull request by sergiy-skalelabs**
- improved SNB parallel startup safety
- missprint fixes

### CODE REVIEW

**Review by payvint — APPROVED** [review]

File: npms/skale-cc/cc.js [hunk 1]
```diff
@@ -169,7 +169,7 @@ const g_arrRainbowParts = [
     g_map_color_definitions.fgMagenta
 ];
 
-function raibow_part( s, i ) {
+function rainbow_part( s, i ) {
```
Comment (DmytroNazarenko) [inline npms/skale-cc/cc.js:172]: Is there any usage of these?

File: npms/skale-ima/index.js [hunk 2]
```diff
@@ -262,8 +262,8 @@ async function get_web3_transactionCount( details, cntAttempts, w3, address, par
     }
     ++ idxAttempt;
     while( ret === "" && idxAttempt <= cntAttempts ) {
-        const isOnlone = rpcCall.check_url( u, nWaitStepMilliseconds );
-        if( ! isOnlone ) {
+        const isOnLine = rpcCall.check_url( u, nWaitStepMilliseconds );
```
Comment (DmytroNazarenko) [inline npms/skale-ima/index.js:265]: Is there any other usages?

File: npms/skale-observer/observer_worker.js [hunk 3]
```diff
@@ -38,14 +38,17 @@ const skale_observer = require( "./observer.js" );
 const g_url = "skale_observer_worker_server";
 
 parentPort.on( "message", jo => {
+    // console.log( cc.attention( ">>>> TRACE - SNB WORKER MESSAGE >>>>" ), cc.j( jo ) );
```
Comment (DmytroNazarenko) [inline npms/skale-observer/observer_worker.js:41]: Why do you need these comments?

File: npms/skale-ima/index.js [hunk 4]
```diff
@@ -327,8 +327,8 @@ async function get_web3_transactionReceipt( details, cntAttempts, w3, txHash, re
     }
     ++ idxAttempt;
     while( txReceipt === "" && idxAttempt <= cntAttempts ) {
-        const isOnlone = rpcCall.check_url( u, nWaitStepMilliseconds );
-        if( ! isOnlone ) {
+        const isOnLine = rpcCall.check_url( u, nWaitStepMilliseconds );
```
Comment (DmytroNazarenko) [inline npms/skale-ima/index.js:330]: Is there any other usages?

File: npms/skale-observer/observer.js [hunk 5]
```diff
@@ -451,14 +454,14 @@ async function cache_schains( strChainNameConnectedTo, w3_main_net, w3_s_chain,
                 cc.debug( "Connected " ) + cc.attention( "S-Chains" ) + cc.debug( " cache was updated in this thread: " ) +
                 cc.j( g_arr_schains_cached ) + "\n" );
         }
-        if( opts.fn_chache_changed )
-            opts.fn_chache_changed( g_arr_schains_cached, null ); // null - no error
+        if( opts.fn_cache_changed )
```
Comment (DmytroNazarenko) [inline npms/skale-observer/observer.js:457]: Where is the declaration of `fn_cache_changed`?

Comment (sergiy-skalelabs) [reply npms/skale-observer/observer.js:457]: `opts.fn_cache_changed ` is future use, not used now

File: npms/skale-observer/observer_worker.js (hunk 3 above)
Comment (sergiy-skalelabs) [reply npms/skale-observer/observer_worker.js:41]: no, can remove it

File: npms/skale-ima/index.js (hunk 2 above)
Comment (sergiy-skalelabs) [reply npms/skale-ima/index.js:265]: yes

File: npms/skale-ima/index.js (hunk 4 above)
Comment (sergiy-skalelabs) [reply npms/skale-ima/index.js:330]: yes

File: npms/skale-cc/cc.js (hunk 1 above)
Comment (sergiy-skalelabs) [reply npms/skale-cc/cc.js:172]: only once somwere in `--help` handler

**Review by DmytroNazarenko — APPROVED** [review]

**Review by kladkogex — APPROVED** [review]

**Pull request merged**